feat(manager/pip-compile): Add support for non-default package repos to the pip-compile manager (#26853)
renovatebot/renovate: docs/usage/getting-started/private-packages.md

name = "pypi"
```

### pip-compile

The pip-compile manager extracts `--index-url` and `--extra-index-url` directives from its input file.
These URLs pull credentials from matching `hostRules` blocks and pass appropriate `--index-url` or `--extra-index-url` flags to `pip-compile` with those credentials while generating the output file.

> ``` title="requirements.in:"

Reviewer: ```suggestion
``` title="requirements.in"
```

We don't need the double colon anymore.

name = "pypi"
```

### pip-compile

> The pip-compile manager will extract `--index-url` and `--extra-index-url` directives from its input file. These URLs will pull credentials from matching `hostRules` blocks and pass appropriate `--index-url` or `--extra-index-url`
> flags to `pip-compile` with those credentials while generating the output file.
> 

Reviewer: ```suggestion
The pip-compile manager extracts `--index-url` and `--extra-index-url` directives from its input file.
These URLs pull credentials from matching `hostRules` blocks and pass appropriate `--index-url` or `--extra-index-url` flags to `pip-compile` with those credentials while generating the output file.
```

Let's start small and follow the one sentence per line convention. I find this paragraph hard to understand, so it needs a full rewrite.

> requirements.in:
> 
> ```

Reviewer: ```suggestion
``` title="requirements.in:"
```

We can put a title in the codeblocks. This title will then show on the published page in the Renovate docs site. 😉 

https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-a-title
